File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/di/jodel/JodelSubComponent.java
package de.jbamberger.offlineclient.di.jodel;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;
import de.jbamberger.offlineclient.ui.jodel.JodelActivity;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@JodelActivityScope
@Subcomponent(modules = {JodelModule.class/*, NetModule.class*/})
public interface JodelSubComponent extends AndroidInjector<JodelActivity> {

    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<JodelActivity> {
    }
}

File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/di/jodel/JodelSubComponent.kt
package de.jbamberger.offlineclient.di.jodel

import dagger.Subcomponent
import dagger.android.AndroidInjector
import de.jbamberger.offlineclient.ui.jodel.JodelActivity

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@JodelActivityScope
@Subcomponent(modules = arrayOf(JodelModule::class/*, NetModule.class*/))
interface JodelSubComponent : AndroidInjector<JodelActivity> {

    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<JodelActivity>()
}
